src/sentry/static/sentry/app/components/activity/noteInput.jsx

```diff
@@ -182,10 +191,26 @@ const NoteInput = React.createClass({
     this.finish();
   },
 
+  onAdd(id, display) {
+    let mentions = this.state.mentions;
+    mentions.push([id, display]);
+    this.setState({mentions: mentions});
+  },
+
   finish() {
     this.props.onFinish && this.props.onFinish();
   },
 
+  finalMentions() {
+    let results = [];
+    this.state.mentions.forEach(item => {
+      if (this.state.value.indexOf(item[1]) !== -1) {
+        results.push(item[0]);
+      }
+    });
+    return results;
```

Comment:
Some feedback:

1. I have no idea what `item[0]` or `item[1]` is here. Comment maybe or use named temp variables?
2. This is nitpicky, but I prefer chaining `Array.prototype.filter` and `Array.prototype.map` because I think it communicates better what you're trying to do.

```javascript
return this.state.mentions
  .filter(item => this.state.value.indexOf(item[1])) // reduce list of mentions only to those found in text
  .map(item => item[0]);